server/src/test/java/org/apache/druid/discovery/BaseNodeRoleWatcherTest.java

```diff
@@ -116,6 +116,19 @@ public void testGeneralUseSimulation()
     assertListener(listener3, true, nodesAdded, nodesRemoved);
   }
 
+  @Test(timeout = 60_000L)
+  public void testTimeOutAfterInitialization() throws InterruptedException
+  {
+    BaseNodeRoleWatcher nodeRoleWatcher = new BaseNodeRoleWatcher(
+        Execs.scheduledSingleThreaded("BaseNodeRoleWatcher"),
+        NodeRole.BROKER
+    );
+    TestListener listener = new TestListener();
+    nodeRoleWatcher.registerListener(listener);
+    Thread.sleep(32_000);
```

Comment:
We want tests to run quickly, so a 32s sleep is too long. You can handle this by introducing a parameter we can use for testing that controls the timeout, and set it shorter during the unit test.

Reply:
Done